spec/plainyearmonth.html

```diff
@@ -611,12 +611,12 @@ <h1>
         1. If ? CalendarEquals(_calendar_, _other_.[[Calendar]]) is *false*, throw a *RangeError* exception.
         1. Let _settings_ be ? GetDifferenceSettings(_operation_, _options_, ~date~, &laquo; *"week"*, *"day"* &raquo;, *"month"*, *"year"*).
         1. Let _fieldNames_ be ? CalendarFields(_calendar_, « *"monthCode"*, *"year"* »).
-        1. Let _otherFields_ be ? PrepareTemporalFields(_other_, _fieldNames_, «»).
-        1. Perform ! CreateDataPropertyOrThrow(_otherFields_, *"day"*, *1*<sub>𝔽</sub>).
-        1. Let _otherDate_ be ? CalendarDateFromFields(_calendar_, _otherFields_).
         1. Let _thisFields_ be ? PrepareTemporalFields(_yearMonth_, _fieldNames_, «»).
         1. Perform ! CreateDataPropertyOrThrow(_thisFields_, *"day"*, *1*<sub>𝔽</sub>).
         1. Let _thisDate_ be ? CalendarDateFromFields(_calendar_, _thisFields_).
+        1. Let _otherFields_ be ? PrepareTemporalFields(_other_, _fieldNames_, «»).
+        1. Perform ! CreateDataPropertyOrThrow(_otherFields_, *"day"*, *1*<sub>𝔽</sub>).
+        1. Let _otherDate_ be ? CalendarDateFromFields(_calendar_, _otherFields_).
         1. Let _untilOptions_ be OrdinaryObjectCreate(*null*).
         1. Perform ? CopyDataProperties(_untilOptions_, _settings_.[[Options]], « »).
```

Comment:
It is curious that this step can throw; is that a bug in failure to properly consume _options_ earlier on?

Reply:
I don't think it's a bug - we have to perform CopyDataProperties on _options_ at some point, so that we can pass an options object to CalendarDateUntil with a modified `largestUnit` property. _options_ is a user-provided object so it may have accessor properties.

Reply:
Right, I'm saying that the bug is consuming it so deep in the spec algorithms rather than in initial processing—and in particular in performing CopyDataProperties _after_ GetDifferenceSettings has reads several properties on its own.

Reply:
I could see this going either way - it'd be good to eliminate the duplicate reads, but on the other hand we'd introduce an inconsistency where we'd copy the options object in the since/until methods of PlainDate, PlainDateTime, PlainYearMonth, and ZonedDateTime; and not copy it in the since/until methods of PlainTime and Instant.

Reply:
Wouldn't options be copied in _all_ since/until methods?

Reply:
Only the ones where a modified options object needs to be passed on to `calendar.dateUntil()`. Otherwise, we just read only the needed properties in alphabetical order.

Reply:
Well, either that serves as explanation of the inconsistency or we'd copy options in all since/until methods even though some technically don't need it.

Reply:
Do you have a concrete proposal for this? If we can manage to write that additional change before TC39's review deadline, I wouldn't be opposed to including it here. Otherwise I think it can be done as part of #2289 or left as is.

Reply:
Concretely, I'm proposing `1. Perform ? CopyDataProperties(_optionsCopy_, _options_, « »).` in every DifferenceTemporal\<Type> operation immediately after validating _other_ and use of _optionsCopy_ rather than _options_ in every subsequent step. But I'm also comfortable with a compromise in which that change is omitted in DifferenceTemporalInstant and DifferenceTemporalPlainTime, ideally replaced with a NOTE step explaining its omission.

I agree that it is logically part of the fixes for #2289, but will still try to put up a narrow PR since it has been specifically identified.